WordPressAuthenticator/NUX/NUXViewControllerBase.swift

```diff
@@ -286,14 +286,46 @@ extension NUXViewControllerBase where Self: UIViewController, Self: UIViewContro
     private func setupNotificationsIndicator() {
         helpNotificationIndicator.isHidden = true
 
-        NotificationCenter.default.addObserver(forName: .wordpressSupportNotificationReceived, object: nil, queue: nil) { [weak self] _ in
+        wordpressSupportNotificationReceivedObserver = NotificationCenter.default.addObserver(forName: .wordpressSupportNotificationReceived, object: nil, queue: nil) { [weak self] _ in
             self?.refreshSupportNotificationIndicator()
         }
-        NotificationCenter.default.addObserver(forName: .wordpressSupportNotificationCleared, object: nil, queue: nil) { [weak self] _ in
+        wordpressSupportNotificationClearedObserver = NotificationCenter.default.addObserver(forName: .wordpressSupportNotificationCleared, object: nil, queue: nil) { [weak self] _ in
             self?.refreshSupportNotificationIndicator()
         }
     }
 
+    private var wordpressSupportNotificationReceivedObserver: NSObjectProtocol? {
+        get {
+
+            objc_getAssociatedObject(self, &wordpressSupportNotificationReceivedKey) as? NSObjectProtocol
+        }
+        set {
+            objc_setAssociatedObject(self, &wordpressSupportNotificationReceivedKey, newValue, .OBJC_ASSOCIATION_RETAIN)
+        }
+    }
+
+    private var wordpressSupportNotificationClearedObserver: NSObjectProtocol? {
+        get {
+            objc_getAssociatedObject(self, &wordpressSupportNotificationClearedKey) as? NSObjectProtocol
+        }
+        set {
+            objc_setAssociatedObject(self, &wordpressSupportNotificationClearedKey, newValue, .OBJC_ASSOCIATION_RETAIN)
+        }
+    }
```

Comment:
Thanks! Yes, I don't like adding this Obj-C magic to the code, I didn't want to refactor the structure of the code too much though. I will check your approach, maybe it will be a sweet spot between avoiding the Obj-C runtime and having to refactor too much in the original code.

Reply:
I looked at the change and it seems like a good approach (not one I would have thought of) 👍 
`addNotificationObserver` caught my eye because it's storing the observer (after its added), despite the name being being **add**. Definitely not an issue, just an observation and no need (from my side) to change it.

Reply:
I forgot to mention, the approach you suggested wasn't acceptable to the compiler. If we wanted to set observer value, we had to make all the methods inside the extension mutable. Also, it prevented from calling those methods in`deinit`.  

> addNotificationObserver caught my eye because it's storing the observer (after its added), despite the name being being add. Definitely not an issue, just an observation and no need (from my side) to change it.

Yes, probably not the best naming. Due to the `mutable` issue I described above, I need to make the value-changing operations from inside the viewControllers. I thought using array instead of separate observer vars will be a cleaner approach.

Reply:
> Do you recall if you made the protocol class-bound? 

No, I haven't, that's a good point, I missed that from the mutable hint of the compiler 🥲 Yes, then that would've worked.

Reply:
Ok 👍